symengine/integer.h

```diff
@@ -152,12 +152,13 @@ struct RCPIntegerKeyLess
         return false;
     }
 };
-//! \return RCP<const Integer> from `int`
-inline RCP<const Integer> integer(int i)
+//! \return RCP<const Integer> from integral values
+template<typename T> inline
+typename std::enable_if<std::is_integral<T>::value, RCP<const Integer>>::type
```

Comment:
Do we still need this `enable_if` in here? It seems it would work without it as well.